config/config.go

```diff
@@ -123,13 +117,31 @@ func (tcf *TerragruntConfigFile) convertToTerragruntConfig(terragruntOptions *op
 		tcf.ExtraArgs = append(tcf.ExtraArgs, tcf.Terraform.LegacyExtraArgs...)
 	}
 
+	if !tcf.InputsHclDefinition.IsNull() {
+		if err := util.FromCtyValue(tcf.InputsHclDefinition, &tcf.Inputs); err != nil {
+			return nil, err
+		}
+	}
+
 	tcf.ExtraArgs.init(tcf)
 	tcf.ExtraCommands.init(tcf)
 	tcf.ImportFiles.init(tcf)
 	tcf.ImportVariables.init(tcf)
 	tcf.ApprovalConfig.init(tcf)
 	tcf.PreHooks.init(tcf)
 	tcf.PostHooks.init(tcf)
+	tcf.RunConditions = RunConditions{}
+	for _, condition := range tcf.RunConditionsHclDefinition {
+		var conditionValue map[string]interface{}
+		if err := util.FromCtyValue(condition.Condition, &conditionValue); err != nil {
+			return nil, err
+		}
+		if condition.IgnoreIfTrue {
```

Comment:
Yes

Reply:
I feel it makes more sense to have `run_if` and `ignore_if` as two separate blocks, that the user is only allowed to add one of them, and they add the conditions inside.

Reply:
Thing is you can have multiple different run or ignore conditions

```
run_conditions {
  condition = {
     region = "us-east-1"
  }
}

run_conditions {
  condition = {
     env = "dev"
  }
}
```
would mean that it runs if either region is us-east-1 or env is dev

```
run_conditions {
  condition = {
     region = "us-east-1"
     env = "dev"
  }
}
```
means that region and env have to match

Is that what you meant?

Reply:
OK no I get it. I was thinking of this actually:
```
run_conditions {
  run_if {
    region = "us-east-1"
  }
}
run_conditions {
  ignore_if {
    region = "us-east-1"
  }
}
```
instead of having a boolean parameter to indicate whether to run or ignore. But it's not a big deal

Reply:
Yeah, this won't work though. because in your example, `run_if` is unstructured data (you can put anything in there), so it can't be a block.

How about ```
run_conditions {
  run_if = {
    region = "us-east-1"
  }
}
run_conditions {
  ignore_if = {
    region = "us-east-1"
  }
}
```